src/util/std_expr.h

```diff
@@ -4564,4 +4564,64 @@ inline array_comprehension_exprt &to_array_comprehension_expr(exprt &expr)
   return ret;
 }
 
+/// An expression for a virtual function
+class virtual_function_exprt : public nullary_exprt
+{
+public:
+  explicit virtual_function_exprt(
+    typet _type,
+    irep_idt component_name,
+    irep_idt class_name,
+    irep_idt base_name,
+    irep_idt identifier)
+    : nullary_exprt(ID_virtual, std::move(_type))
+  {
+    set(ID_component_name, std::move(component_name));
+    set(ID_C_class, std::move(class_name));
+    set(ID_C_base_name, std::move(base_name));
+    set(ID_identifier, std::move(identifier));
+  }
+
+  const irep_idt &get_component_name() const
+  {
+    return get(ID_component_name);
+  }
+
+  const irep_idt &get_class_name() const
+  {
+    return get(ID_C_class);
+  }
+
+  const irep_idt &get_base_name() const
+  {
+    return get(ID_C_base_name);
+  }
+
+  const irep_idt &get_identifier() const
+  {
+    return get(ID_identifier);
+  }
+};
+
+/// \brief Cast an exprt to a \ref virtual_function_exprt
+///
+/// \a expr must be known to be \ref virtual_function_exprt.
+///
+/// \param expr: Source expression
+/// \return Object of type \ref virtual_function_exprt
+inline const virtual_function_exprt &to_virtual_function_expr(const exprt &expr)
+{
+  PRECONDITION(expr.id() == ID_virtual_function);
```

Comment:
Consider `PRECONDITION(can_cast_expr<virtual_function_exprt>(expr));`

Reply:
That's not how the other ones are done, so I did it like this for consistency